flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -331,6 +346,24 @@ private void registerInputMetrics(MetricGroup inputGroup, MetricGroup buffersGro
 		buffersGroup.gauge(METRIC_INPUT_POOL_USAGE, new InputBufferPoolUsageGauge(inputGates));
 	}
 
+	/**
+	 * Update consuming gate with newly available partition.
+	 *
+	 * @param partitionInfo telling where the partition can be retrieved from
+	 * @throws IOException IO problem by the update
+	 * @throws InterruptedException potentially blocking operation was interrupted
+	 * @throws PartitionException the input gate with the id from the partitionInfo is not found
+	 */
+	public void updatePartitionInfo(PartitionInfo partitionInfo)
+		throws IOException, InterruptedException, PartitionException {
```

Comment:
Personally I'd prefer to put `throws` on the same line as `)`. The problem is that it now looks as if the `throws` line belongs to the body. Just personal taste, though.